src/librustc/mir/mod.rs

```diff
@@ -1674,6 +1674,10 @@ impl<'tcx> Statement<'tcx> {
     /// Changes a statement to a nop. This is both faster than deleting instructions and avoids
     /// invalidating statement indices in `Location`s.
     pub fn make_nop(&mut self) {
+        // `Statement` contributes significantly to peak memory usage. Make
+        // sure it doesn't get bigger.
+        static_assert!(STATEMENT_IS_AT_MOST_56_BYTES: mem::size_of::<Statement<'_>>() <= 56);
```

Comment:
Why is this inside some random function, as opposed to, say, next to `enum Statement`?

Reply:
I didn't realize `static_assert!` was valid at the top level. I'll fix it in another PR, thanks for the tip.